src/transformers/trainer.py

```diff
@@ -197,13 +236,19 @@ def get_eval_dataloader(self, eval_dataset: Optional[Dataset] = None) -> DataLoa
 
         eval_dataset = eval_dataset if eval_dataset is not None else self.eval_dataset
 
-        sampler = get_tpu_sampler(eval_dataset) if is_tpu_available() else None
+        if is_tpu_available():
+            sampler = SequentialDistributedSampler(
+                eval_dataset, num_replicas=xm.xrt_world_size(), rank=xm.get_ordinal()
+            )
+        elif self.args.local_rank != -1:
+            sampler = SequentialDistributedSampler(eval_dataset)
+        else:
+            sampler = SequentialSampler(eval_dataset)
 
         data_loader = DataLoader(
             eval_dataset,
             sampler=sampler,
             batch_size=self.args.eval_batch_size,
-            shuffle=False,
```

Comment:
Again, why remove shuffling for all eval datasets? This obviously still works but just wondering if there's ever a situation where eval dataset may need to be shuffled.

Reply:
Now that I think of this, for testing/evaluation you don't need a shuffled dataset anyway so it doesn't matter. Shuffling data is a means to ensure that you don't unknowingly put datapoints together that are related in one way and that give a poor average representation of the whole dataset _during training_. For eval/text it doesn't matter.